tests/src/androidTest/java/com/squareup/duktape/DuktapeBindTest.java

```diff
@@ -0,0 +1,383 @@
+/*
+ * Copyright (C) 2016 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.duktape;
+
+import android.support.test.runner.AndroidJUnit4;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.Date;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.junit.Assert.fail;
+
+@RunWith(AndroidJUnit4.class)
+public final class DuktapeBindTest {
+  private Duktape duktape;
+
+  @Before public void setUp() {
+    duktape = Duktape.create();
+  }
+
+  @After public void tearDown() {
+    duktape.close();
+  }
+
+  @Test public void bindNonInterface() {
+    try {
+      duktape.bind("s", String.class, "foo");
+      fail();
+    } catch (UnsupportedOperationException expected) {
+      assertThat(expected)
+          .hasMessage("Only interfaces can be bound. Received: class java.lang.String");
+    }
+  }
+
+  interface TestInterface {
+    String getValue();
+  }
+
+  @Test public void callMethodOnJavaObject() {
+    duktape.bind("value", TestInterface.class, new TestInterface() {
+      @Override
+      public String getValue() {
+        return "8675309";
+      }
+    });
+    assertThat(duktape.evaluate("value.getValue();")).isEqualTo("8675309");
+  }
+
+  @Test public void callMissingMethodOnJavaObjectFails() {
+    duktape.bind("value", TestInterface.class, new TestInterface() {
+      @Override
+      public String getValue() {
+        return "bar";
+      }
+    });
+    try {
+      duktape.evaluate("value.increment()");
+      fail();
+    } catch (DuktapeException expected) {
```

Comment:
Yeah, the difference is here the `DuktapeException` is wrapping something thrown by Duktape.

I guess that's another possibility: `DuktapeException` is something thrown by Duktape and we continue to use `NoSuchMethodError` when proxying, since it's sort of a Java-side thing?  I don't really feel strongly either way, though.